sroka/api/s3_connection/s3_connection_api.py

```diff
@@ -28,15 +29,25 @@ def _download_data(key_prefix, s3, bucket_name, prefix, sep):
             df_list.append(pd.read_csv(data, error_bad_lines=False, warn_bad_lines=False, sep=sep))
         except UnicodeDecodeError:
             df_list.append(pq.read_pandas(data).to_pandas())
+        except EmptyDataError:
+            print('File is empty')
+            return pd.DataFrame([])
 
     else:
         bucket = s3.Bucket(bucket_name)
 
         try:
             for file in bucket.objects.filter(Prefix=key_prefix):
                 if 'SUCCESS' not in file.key:
-                    data = StringIO(str(file.get()['Body'].read(), 'utf-8'))
-                    df_list.append(pd.read_csv(data, error_bad_lines=False, warn_bad_lines=False, sep=sep))
+                    tmp = StringIO(str(file.get()['Body'].read(), 'utf-8'))
+                    try:
+                        data = pd.read_csv(tmp, error_bad_lines=False, warn_bad_lines=False, sep=sep)
+                        df_list.append(data)
+                    except EmptyDataError:
+                        if skip_empty_files is False:
+                            print('One or more files are empty')
```

Comment:
At this point it's the current file, isn't it? Therefore, I suggest also informing a user about a concrete file which was empty and caused the exception.

Reply:
Yes it's a good point thank you!